IPython/core/display.py

```diff
@@ -139,6 +142,124 @@ def _new_id():
     """Generate a new random text id with urandom"""
     return b2a_hex(os.urandom(16)).decode('ascii')
 
+class RecursiveObject:
+    """
+    Default recursive object that provides a recursion repr if needed.
+
+    You may register a formatter for this object that will be call when
+    recursion is reached.
+    """
+
+    def __init__(self, already_seen):
+        self.seen = already_seen
+        pass
+
+    def __repr__(self):
+        return '<recursion ... {}>'.format(str(self.seen))
+
+    def _repr_html_(self):
+        import html
+        return '&lt;recursion ... {}&gt;'.format(html.escape(str(self.seen)))
+
+
+
+DataMetadata = namedtuple('DataMetadata', ('data','metadata'))
```

Comment:
You were asking about the issue of conveniently retrieving mime-keys with the `('data', 'metadata')` tuple. One option is a method on this object:

```python
def get_mime(self, mimetype):
    """Extract a single mime-type representation from a mimebundle"""
    if mimetype not in self.data:
        return (None, None)
    else:
        return self.data[mimetype], self.metadata.get(mimetype)
```

This could also be wrapped in a single function:

```python
def get_repr_mime(obj, mimetype):
    """Get the representation of an object as a single mime-type"""
    return get_repr_mimebundle(obj).get_mime(mimetype)
```

A `metadata=False` flag could also prevent it from returning the metadata part, if you want to avoid getting the first item for the data part.